### ISSUE TITLE
[magic-enum] v 0.9.6 build failure

### ISSUE THREAD

**Issue by anders-wind** [microsoft/vcpkg#40429]
### Operating system

linux

### Compiler

GCC, Clang

### Steps to reproduce the behavior

```Shell
`vcpkg install magic-enum`
```

### Failure logs

```
Computing installation plan...
A suitable version of cmake was not found (required v3.29.2).
Downloading https://github.com/Kitware/CMake/releases/download/v3.29.2/cmake-3.29.2-linux-x86_64.tar.gz
Extracting cmake...
The following packages will be built and installed:
    magic-enum:x64-linux@0.9.6
  * vcpkg-cmake:x64-linux@2024-04-23
  * vcpkg-cmake-config:x64-linux@2024-05-23
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-linux...
Compiler found: /usr/bin/c++
Restored 0 package(s) from /home/anderswind/.cache/vcpkg/archives in 1.82 ms. Use --debug to see more details.
Installing 1/3 vcpkg-cmake:x64-linux@2024-04-23...
Building vcpkg-cmake:x64-linux@2024-04-23...
-- Installing: /home/anderswind/projects/vcpkg/packages/vcpkg-cmake_x64-linux/share/vcpkg-cmake/vcpkg_cmake_configure.cmake
-- Installing: /home/anderswind/projects/vcpkg/packages/vcpkg-cmake_x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake
-- Installing: /home/anderswind/projects/vcpkg/packages/vcpkg-cmake_x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake
-- Installing: /home/anderswind/projects/vcpkg/packages/vcpkg-cmake_x64-linux/share/vcpkg-cmake/vcpkg-port-config.cmake
-- Installing: /home/anderswind/projects/vcpkg/packages/vcpkg-cmake_x64-linux/share/vcpkg-cmake/copyright
-- Performing post-build validation
Stored binaries in 1 destinations in 2.4 ms.
Elapsed time to handle vcpkg-cmake:x64-linux: 18.8 ms
vcpkg-cmake:x64-linux package ABI: 2369fe99df15195caeb4bd124934fc573ac06fafcad4b19eeb3f3c6e12a6bc3f
Installing 2/3 vcpkg-cmake-config:x64-linux@2024-05-23...
Building vcpkg-cmake-config:x64-linux@2024-05-23...
-- Installing: /home/anderswind/projects/vcpkg/packages/vcpkg-cmake-config_x64-linux/share/vcpkg-cmake-config/vcpkg_cmake_config_fixup.cmake
-- Installing: /home/anderswind/projects/vcpkg/packages/vcpkg-cmake-config_x64-linux/share/vcpkg-cmake-config/vcpkg-port-config.cmake
-- Installing: /home/anderswind/projects/vcpkg/packages/vcpkg-cmake-config_x64-linux/share/vcpkg-cmake-config/copyright
-- Skipping post-build validation due to VCPKG_POLICY_EMPTY_PACKAGE
Stored binaries in 1 destinations in 2.77 ms.
Elapsed time to handle vcpkg-cmake-config:x64-linux: 18.7 ms
vcpkg-cmake-config:x64-linux package ABI: 96367de84075fe3907c23fb9d2d22cc6ff90e42cb4755dae86bd467ca794f6ed
Installing 3/3 magic-enum:x64-linux@0.9.6...
Building magic-enum:x64-linux@0.9.6...
-- Downloading https://github.com/Neargye/magic_enum/archive/v0.9.6.tar.gz -> Neargye-magic_enum-v0.9.6.tar.gz...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Trying to load bundleconfig from /home/anderswind/projects/vcpkg/vcpkg-bundle.json
[DEBUG] Failed to open: /home/anderswind/projects/vcpkg/vcpkg-bundle.json
[DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[DEBUG] Metrics enabled.
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Feature flag 'dependencygraph' unset
[DEBUG] 1000: execute_process(curl --fail -L https://github.com/Neargye/magic_enum/archive/v0.9.6.tar.gz --create-dirs --output /home/anderswind/projects/vcpkg/downloads/Neargye-magic_enum-v0.9.6.tar.gz.4160505.part)
[DEBUG] 1000: cmd_execute_and_stream_data() returned 0 after   681108 us
[DEBUG] Trying to hash /home/anderswind/projects/vcpkg/downloads/Neargye-magic_enum-v0.9.6.tar.gz.4160505.part
[DEBUG] /home/anderswind/projects/vcpkg/downloads/Neargye-magic_enum-v0.9.6.tar.gz.4160505.part has hash fa3792e9c69c57e437d6325b7659cf25e6cbdda3326ffeaf0411d9838822095b15306f322ad2ebd5ec68b905ef2aed08880d2507e7d8a8559ab3c5be6c8ce99c
error: Missing Neargye-magic_enum-v0.9.6.tar.gz and downloads are blocked by x-block-origin.
error: File does not have the expected hash:
url: https://github.com/Neargye/magic_enum/archive/v0.9.6.tar.gz
File: /home/anderswind/projects/vcpkg/downloads/Neargye-magic_enum-v0.9.6.tar.gz.4160505.part
Expected hash: 6154c816446e115f3b164df79ab8d8088eb76b632ee3fdc82ea17cc7ae8d04652c83e5cc587c2c4b334889904b101ba08a04c5837103af260768e93df17cc263
Actual hash: fa3792e9c69c57e437d6325b7659cf25e6cbdda3326ffeaf0411d9838822095b15306f322ad2ebd5ec68b905ef2aed08880d2507e7d8a8559ab3c5be6c8ce99c
[DEBUG] /source/src/vcpkg/base/downloads.cpp(1030):
[DEBUG] Time in subprocesses: 681108us
[DEBUG] Time in parsing JSON: 4us
[DEBUG] Time in JSON reader: 0us
[DEBUG] Time in filesystem: 212us
[DEBUG] Time in loading ports: 0us
[DEBUG] Exiting after 683 ms (682397us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):

      Failed to download file with error: 1
      If you are using a proxy, please check your proxy setting. Possible causes are:

      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.

      2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
         set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
         The value set by your proxy might be wrong, or have same `https://` prefix issue.

      3. Your proxy's remote server is out of service.

      If you've tried directly download the link, and believe this is not a temporary
      download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
      to report this upstream download server failure.


Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
  ports/magic-enum/portfile.cmake:1 (vcpkg_from_github)
  scripts/ports.cmake:192 (include)


error: building magic-enum:x64-linux failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle magic-enum:x64-linux: 951 ms
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+magic-enum
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[magic-enum]+Build+error+on+x64-linux&body=Copy+issue+body+from+%2Fhome%2Fanderswind%2Fprojects%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
You can also submit an issue by running (GitHub CLI must be installed):
  gh issue create -R microsoft/vcpkg --title "[magic-enum] Build failure on x64-linux" --body-file /home/anderswind/projects/vcpkg/installed/vcpkg/issue_body.md
```

### Additional context

https://github.com/Neargye/magic_enum/issues/373

**Comment by anders-wind** [comment]
Im creating a hotfix PR
